loki/transformations/inline.py

```diff
@@ -184,6 +196,15 @@ def map_inline_call(self, expr, *args, **kwargs):
             # We still need to recurse and ensure re-scoping
             return super().map_inline_call(expr, *args, **kwargs)
 
+        # if it is an inline call to a Statement Function
+        if isinstance(expr.routine, StatementFunction):
+            function = expr.routine
+            v_result = expr.routine.variable
+            # Substitute all arguments through the elemental body
+            arg_map = dict(zip(function.arguments, expr.parameters))
```

Comment:
Just wondering: Wouldn't `arg_map = dict(expr.arg_iter())` work here, and also below for "regular" inline calls? That should also take care of keyword arguments.
(Cc @mlange05)